zrouting/routinglookup_test.go

```diff
@@ -0,0 +1,59 @@
+package zrouting
+
+import (
+	"testing"
+
+	"gotest.tools/assert"
+)
+
+type pathFilterTest struct {
+	path     []uint32
+	filter   PathFilter
+	expected []uint32
+}
+
+// 64512 - 65534
+
+func TestIBGPFilter(t *testing.T) {
+	tests := []pathFilterTest{
+		pathFilterTest{
+			path:     []uint32{1, 2, 3, 4},
+			filter:   IdentityPathFilter,
+			expected: []uint32{1, 2, 3, 4},
+		},
+		pathFilterTest{
+			path:     []uint32{1, 65000, 38},
+			filter:   InternalBGPPathFilter(38),
+			expected: []uint32{1, 38},
+		},
+		pathFilterTest{
+			path:     []uint32{1, 65000},
+			filter:   InternalBGPPathFilter(38),
+			expected: []uint32{1, 38},
+		},
+		pathFilterTest{
+			path:     []uint32{1, 65000, 64512, 65118, 38, 27},
```

Comment:
I don't think internal and external ASNs can be mixed. Doesn't seem to make intuitive sense.

Reply:
I agree, but I figured I'd handle it.